### PR TITLE
Websocket chat

### PR DESCRIPTION
**Pull request by teddylo57**

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -43,7 +43,17 @@
     <dependency>
       <groupId>com.sparkjava</groupId>
       <artifactId>spark-core</artifactId>
-      <version>2.7.1</version>
+      <version>2.7.2</version>
+    </dependency>
+    <dependency>
+      <groupId>com.j2html</groupId>
```
Comment (ericliu121187) [inline backend/pom.xml]: Do we need this?

Comment (teddylo57) [reply backend/pom.xml]: not sure if you need it to render html, if not i can delete

File: backend/src/main/java/poker/Api.java [hunk 2]
```diff
@@ -14,6 +15,8 @@
 public class Api {
 
   public static void main(String[] args) {
+    webSocket("/chat", ChatWebSocketHandler.class);
```
[the remark this answers is not in the document]
Comment (teddylo57) [reply backend/src/main/java/poker/Api.java]: yes it works, but i am unable to e2e test without the corresponding js component

File: backend/src/main/java/poker/models/Table.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package poker.models;
+
+import java.util.List;
+
+public class Table {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/Table.java]: Is this all that is needed in the Table class? Does this handle seating and standing for players?

Comment (teddylo57) [reply backend/src/main/java/poker/models/Table.java]: added seating and players leaving the table

File: backend/src/main/java/poker/models/Table.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  public void addPlayer(Player newPlayer) throws Exception{
+    if (players.size() < 8) {
+      players.add(newPlayer);
+    } else {
+      throw new Exception("Table is full");
+    }
+  }
+
+  public void removePlayer(Player deletePlayer) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/Table.java]: is there a better name than `deletePlayer`? that sounds a lot like a function name

**Pull request closed without merging**